mysql/datadog_checks/mysql/statement_samples.py

```diff
@@ -506,16 +503,30 @@ def _collect_plan_for_statement(self, row):
             return None
 
         plan, explain_errors = None, None
-        with closing(self._get_db_connection().cursor()) as cursor:
-            try:
-                plan, explain_errors = self._explain_statement(
-                    cursor, row['sql_text'], row['current_schema'], obfuscated_statement
-                )
-            except Exception as e:
-                self._check.count(
-                    "dd.mysql.statement_samples.error", 1, tags=self._tags + ["error:explain-{}".format(type(e))]
+        if self._is_statement_truncated(row['sql_text']) == StatementTruncationState.truncated:
+            self._check.count(
+                "dd.mysql.statement_samples.error",
+                1,
+                tags=self._tags + ["error:explain-{}".format(DBExplainError.statement_truncated)],
+            )
+            explain_errors = [
+                (
+                    None,
+                    DBExplainError.statement_truncated,
+                    None,
```

Comment:
Like for postgres, could be useful to include several of the relevant config values here:

```
max_digest_length
performance_schema_max_digest_length
performance_schema_max_sql_text_length
```

Some of these can be changed dynamically at runtime though so we could cache them for some time, refreshing periodically.

Reply:
We discussed "offline" and thought of a decent approximation that wouldn't require adding a new query to fetch configs (which brings up having to deal with potentially not having permissions to get those): we're just going to include the `len(row['sql_text'])` in the message.